UserTools/DigitBuilder/DigitBuilder.cpp

```diff
@@ -342,45 +352,126 @@ bool DigitBuilder::BuildMCLAPPDRecoDigit() {
+                        std::cout << "THERE ARE HITS ON LAPPD " << LAPPDId << endl;
+                        currentLAPPD = LAPPDId;                        
+                        fHitLAPPDs->push_back(currentLAPPD);
+                        std::cout << "fHitLAPPDs size: " << fHitLAPPDs->size() << endl;
+                    }
+                    if (striphit == 0) {
+                        //if(v_message<verbosity) ahit.Print(); // << VERY verbose
+                        // an LAPPDHit has adds (global x-y-z) position, (in-tile x-y) local position
+                        // and time psecs
+                        // convert the WCSim coordinates to the ANNIEreco coordinates
+                        // convert the unit from m to cm
+                        pos_reco.SetX(ahit.GetPosition().at(0) * 100. + xshift); //cm
+                        pos_reco.SetY(ahit.GetPosition().at(1) * 100. + yshift); //cm
+                        pos_reco.SetZ(ahit.GetPosition().at(2) * 100. + zshift); //cm
+                        calT = ahit.GetTime();  // 
+                        calT = frand.Gaus(calT, 0.1); // time is smeared with 100 ps time resolution. Harded-coded for now.
+                        calQ = ahit.GetCharge();
+                        if (verbosity > 4) {
+                            std::cout << "LAPPD position (X<Y<Z): " <<
+                                to_string(pos_reco.X()) << "," << to_string(pos_reco.Y()) <<
+                                "," << to_string(pos_reco.Z()) << std::endl;
+                            std::cout << "LAPPD Charge,Time: " << to_string(calQ) << "," <<
+                                to_string(calT) << std::endl;
+                        }
+                        // I found the charge is 0 for all the hits. In order to test the code, 
+                        // here I just set the charge to 1. We should come back to this later. (Jingbo Wang)
+                        calQ = 1.;
+                        digitType = RecoDigit::lappd_v0;
+                        RecoDigit recoDigit(region, pos_reco, calT, calQ, digitType, LAPPDId);
+                        //if(v_message<verbosity) recoDigit.Print();
+                      //make some cuts here. It will be moved to the Hitcleaning tool
+                        if (calT > 40 || calT < -10) continue; // cut off delayed hits
+                        fDigitList->push_back(recoDigit);
+
+                    }
+                    else {
+                        posX = (ahit.GetPosition().at(0) * 100 + xshift) - (det->GetPositionInTank().X() * 100 + xshift);
+                        double dirX = det->GetDetectorDirection().X();
+                        double AngX = std::asin(dirX);
+                        int strip = (int)((posX / (20.* std::sin(AngX) / 28.)) + 14 - 1);
+                        if (strip > 27 || strip < 0) {
+                            Log("warning: LAPPD hit found off LAPPD: ", v_debug, verbosity);
```

Comment:
Warnings should probably be logged with a loglevel of `v_warning` rather than `v_debug`.... ?